### PR TITLE
Feature: Add flag for install location (optional)

### PR DESCRIPTION
**Pull request by ArronaxKP**
This change adds a new flag `-i` or `--install` that will allow you to set where the Terraform binary is downloaded/installed into. Useful for edge cases where you need to share commonly installed versions of Terraform. 

This is in draft as it is built upon this PRs that should be merged first. To save headaches with merge conflicts this PR already includes that commit.

- #295 - Thanks @jukie for the PR. 

Once the above PR has been merged ping me and I can update this branch with latest master.

I have created a similar PR for tgswitch here - https://github.com/warrensbox/tgswitch/pull/140

### CODE REVIEW

**Comment by strowi** [comment]
This would be very useful for CI/CD cases where $HOME/ is not a cache-able path.

**Comment by ArronaxKP** [comment]
For CI/CD is what I wrote this for. It turns out we found a workaround for the issue we were having so I didn't need this int eh end

I created a test binary and published it here if you want it but I am not using it myself, I fell back to the original version by warrensbox. I think its just linux_amd64 binary only though:
https://github.com/ArronaxKP/terraform-switcher/releases/tag/tfswitch-with-install-flag

**Comment by warrensbox** [comment]
We can revisit this after release v1.10

**Comment by ArronaxKP** [comment]
Updated with latest master changes. I have moved away from using tfswitch and tgswitch now. But I wanted to attempt to get this PR updated to as best I can. Since I haven't touched this project in over 9 months, it took a while. I "think" it now works. I tested locally quickly and the `-i` install works however I am using windows so not a real test. I don't have time to fully test it on all platforms. 

If this is still wanted please ping me if you need anything. I will leave this PR in draft as I cannot fully test this locally. Feel free to take the code and create a new branch/PR if that is easier.

File: README.md
Comment (yermulnik) [inline README.md:12]: Please put two trailing periods back — they are line breaks in markdown.

File: README.md
Comment (yermulnik) [inline README.md:21]: Please put two trailing periods back — they are line breaks in markdown.

**Review by yermulnik — COMMENTED** [review]
Thank you for re-visiting this PR.
I do hope you won't be much distracted by re-visiting it once again once the #356 is merged — the #356 brings a lot of refactoring, including the bits touched in this PR.
Thanks in advance.

File: README.md
Comment (ArronaxKP) [reply README.md:12]: Ah sorry, I have VS configured to auto remove trailing whitespace. This always catches me out as I avoid the 2 space trailing line breaks due to this break. I tend to use `<br>` instead if I need it based on [line breaks best practices](https://github.com/mattcone/markdown-guide/blob/master/_basic-syntax/line-breaks.md#line-break-best-practices)

Sorry, will put them back now.

File: lib/install.go
Comment (ArronaxKP) [reply lib/install.go:17]: Actually agree, dir would be better. Will fix this

File: www/docs/usage/commandline.md
Comment (ArronaxKP) [reply www/docs/usage/commandline.md]: Fixed

File: README.md
Comment (ArronaxKP) [reply README.md]: Fixed

File: README.md
Comment (ArronaxKP) [reply README.md]: Fixed

File: README.md
Comment (ArronaxKP) [reply README.md]: Fixed

**Comment by ArronaxKP** [comment]
Resolved all comments. 

If you ping me when https://github.com/warrensbox/terraform-switcher/pull/356 is merged. I will update this again, and re-test it.

**Comment by yermulnik** [comment]
> Resolved all comments.

Thank you.

> If you ping me when #356 is merged. I will update this again, and re-test it.

We'll try out best 🤝

**Comment by yermulnik** [comment]
> > If you ping me when #356 is merged. I will update this again, and re-test it.
> 
> We'll try out best 🤝

@ArronaxKP Pinging you

**Comment by ArronaxKP** [comment]
As promised the PR updated with latest master. Includes the origina `-i/--install` flag changes.

Spotted some other warnings in other files I thought I would fix. 

- Moved homedir function to the `files.go` file inline with all other file functions.
- Fixed markdown spacing issues where there was a missing new line between headers and bullet points.
- Corrected some warnings for booleans based on best practice:
  - [S1008 - Simplify returning boolean expression](https://staticcheck.io/docs/checks#S1008).
  - [S1002 - Omit comparison with boolean constant](https://staticcheck.io/docs/checks#S1002).

My github actions are successful:
image

File: lib/param_parsing/parameters.go
Comment (yermulnik) [inline lib/param_parsing/parameters.go:38]: Option description looks a bit messed up with `-b`, doesn't it? 🤔
Also should it probably mention full install path as example for clarity (as in with `installDir` appended)?

File: www/docs/usage/commandline.md
Comment (yermulnik) [inline www/docs/usage/commandline.md:27]: I'd glue this together (or put two trailing periods otherwise):
```suggestion
You can also set the `TF_VERSION` environment variable to your desired terraform version. For example:
```

File: www/docs/usage/commandline.md
Comment (yermulnik) [inline www/docs/usage/commandline.md:80]: It's sort of misleading to use Warren's username and if user is on Linux or Windows. What do you think of this commit suggestion?
```suggestion
By default `tfswitch` will download the Terraform binary to the user home directory under this path: `$HOME/.terraform.versions`
```

File: lib/param_parsing/parameters.go
Comment (ArronaxKP) [reply lib/param_parsing/parameters.go:38]: Will fix/add this now

File: www/docs/usage/commandline.md
Comment (ArronaxKP) [reply www/docs/usage/commandline.md:27]: Fixing

File: www/docs/usage/commandline.md
Comment (ArronaxKP) [reply www/docs/usage/commandline.md:80]: Fixing

File: www/docs/usage/config-files.md
Comment (yermulnik) [inline www/docs/usage/config-files.md:39]:
```suggestion
1. Create a custom binary path. Ex: `mkdir $HOME/bin`
```

**Review by yermulnik — COMMENTED** [review]
LGTM with minor note below

**Comment by yermulnik** [comment]
Also please fetch `master`

**Comment by yermulnik** [comment]
@warrensbox @MatrixCrawler @crablab @MatthewJohn @jukie Please review.

File: www/docs/usage/config-files.md
Comment (ArronaxKP) [reply www/docs/usage/config-files.md]: Sorry missed that bit fixing.

**Comment by ArronaxKP** [comment]
Updated  with latest master. My Github actions is passing as well

**Review by MatrixCrawler — APPROVED** [review]
LGTM

**Comment by crablab** [comment]
Will look this evening/tomorrow morning.

**Comment by yermulnik** [comment]
> Will look this evening/tomorrow morning.

I'm giving this my "LGTM". Please merge once you get this reviewed and if you approve when you have a chance later. Thanks.

**Review by yermulnik — APPROVED** [review]

File: www/docs/usage/commandline.md
Comment (crablab) [inline www/docs/usage/commandline.md:89]:
```suggestion
**NOTE** - The directory passed in `-i`/`--install` must be created before running `tfswitch`
```

**Review by crablab — APPROVED** [review]

**Pull request merged**
